addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/application-load-balancer.cfn.yml

```diff
@@ -9,52 +9,163 @@ Parameters:
+      SubnetId: !Ref PublicSubnet2
+
+  AppStreamSecurityGroupEgress:
+    Type: AWS::EC2::SecurityGroupEgress
+    Condition: AppStreamEnabled
+    Properties:
+      GroupId: !Ref AppStreamSG
+      DestinationSecurityGroupId: !Ref ALBSecurityGroup
+      Description: 'Allow AppStream egress to ALB'
+      IpProtocol: '-1'
+
+  PrivateSubnet1:
+    Type: AWS::EC2::Subnet
+    Condition: AppStreamEnabled
+    Properties:
+      VpcId: !Ref VPC
+      AvailabilityZone: !Select [0, !GetAZs '']
+      CidrBlock: !Ref PrivateSubnet1Cidr
+      MapPublicIpOnLaunch: false
+      Tags:
+        - Key: Name
+          Value: !Sub ${Namespace} ALB private subnet 1
+
+  PrivateSubnet2:
+    Type: AWS::EC2::Subnet
+    Condition: AppStreamEnabled
+    Properties:
+      VpcId: !Ref VPC
+      AvailabilityZone: !Select [1, !GetAZs '']
+      CidrBlock: !Ref PrivateSubnet2Cidr
+      MapPublicIpOnLaunch: false
+      Tags:
+        - Key: Name
+          Value: !Sub ${Namespace} ALB private subnet 2
+
   ALBListener:
-     Type: AWS::ElasticLoadBalancingV2::Listener
-     Properties:
-       DefaultActions:
-         - Type: fixed-response
-           FixedResponseConfig:
-             ContentType: "text/plain"
-             MessageBody: "Forbidden"
-             StatusCode: "403"
-       LoadBalancerArn:
-           Ref: ApplicationLoadBalancer
-       Port: 443
-       Protocol: HTTPS
-       SslPolicy: ELBSecurityPolicy-2016-08
-       Certificates: 
-          - CertificateArn: !Ref ACMSSLCertARN
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    Properties:
+      DefaultActions:
+        - Type: fixed-response
+          FixedResponseConfig:
+            ContentType: 'text/plain'
+            MessageBody: 'Forbidden'
+            StatusCode: '403'
+      LoadBalancerArn:
+        Ref: ApplicationLoadBalancer
+      Port: 443
+      Protocol: HTTPS
+      SslPolicy: ELBSecurityPolicy-2016-08
+      Certificates:
+        - CertificateArn: !Ref ACMSSLCertARN
+
   ApplicationLoadBalancer:
     Type: AWS::ElasticLoadBalancingV2::LoadBalancer
     Properties:
       Name: !Ref Namespace
-      Scheme: internet-facing # or internal
+      Scheme: !If [AppStreamEnabled, 'internal', 'internet-facing']
       Subnets:
-      - Ref: Subnet1
-      - Ref: Subnet2
-      SecurityGroups: 
-      - Ref: ALBSecurityGroup
+        !If [AppStreamEnabled, [!Ref PrivateSubnet1, !Ref PrivateSubnet2], [!Ref PublicSubnet1, !Ref PublicSubnet2]]
+      SecurityGroups:
+        - Ref: ALBSecurityGroup
+
   ALBSecurityGroup:
     Type: AWS::EC2::SecurityGroup
     Properties:
       SecurityGroupIngress:
-        - CidrIp: "0.0.0.0/0"
-          FromPort: 443
-          ToPort: 443
-          IpProtocol: tcp
+        - !If
+          - AppStreamEnabled
+          - SourceSecurityGroupId: !Ref AppStreamSG
+            IpProtocol: '-1'
+          - CidrIp: '0.0.0.0/0'
+            FromPort: 443
+            ToPort: 443
+            IpProtocol: tcp
```

Comment:
This has to be `0.0.0.0` because the ALB needs to be accessible from internet since we don't know the CIDR range where ALB will be connected from?

Reply:
Correct.